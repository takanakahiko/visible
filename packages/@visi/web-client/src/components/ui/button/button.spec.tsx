import { render } from '@testing-library/react';
import React from 'react';

import {
  Primary,
  PrimaryDisabled,
  PrimaryLarge,
  Secondary,
  SecondaryDisabled,
} from './button.stories';

describe('Input', () => {
  test('primary', () => {
    const { container } = render(<Primary />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('primary disabled', () => {
    const { container } = render(<PrimaryDisabled />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('primary large', () => {
    const { container } = render(<PrimaryLarge />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('primary large', () => {
    const { container } = render(<PrimaryLarge />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('secondary', () => {
    const { container } = render(<Secondary />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('secondary disabled', () => {
    const { container } = render(<SecondaryDisabled />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
