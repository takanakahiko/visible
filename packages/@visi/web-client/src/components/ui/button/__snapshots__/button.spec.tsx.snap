// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input primary 1`] = `
<button
  class="flex rounded-full transition-all font-semibold duration-200 ease-in shadow hover:shadow-md focus:outline-none focus:shadow-outline disabled:cursor-not-allowed disabled:opacity-75 px-4 py-2 text-sm bg-primary-500 text-white hover:bg-primary-600 active:bg-primary-700"
  rel="noreferrer"
  target="_blank"
>
  <span>
    text
  </span>
</button>
`;

exports[`Input primary disabled 1`] = `
<button
  class="flex rounded-full transition-all font-semibold duration-200 ease-in shadow hover:shadow-md focus:outline-none focus:shadow-outline disabled:cursor-not-allowed disabled:opacity-75 px-4 py-2 text-sm bg-primary-500 text-white hover:bg-primary-600 active:bg-primary-700"
  disabled=""
  rel="noreferrer"
  target="_blank"
>
  <span>
    text
  </span>
</button>
`;

exports[`Input primary large 1`] = `
<button
  class="flex rounded-full transition-all font-semibold duration-200 ease-in shadow hover:shadow-md focus:outline-none focus:shadow-outline disabled:cursor-not-allowed disabled:opacity-75 px-5 py-3 text-base bg-primary-500 text-white hover:bg-primary-600 active:bg-primary-700"
  rel="noreferrer"
  target="_blank"
>
  <span>
    text
  </span>
</button>
`;

exports[`Input primary large 2`] = `
<button
  class="flex rounded-full transition-all font-semibold duration-200 ease-in shadow hover:shadow-md focus:outline-none focus:shadow-outline disabled:cursor-not-allowed disabled:opacity-75 px-5 py-3 text-base bg-primary-500 text-white hover:bg-primary-600 active:bg-primary-700"
  rel="noreferrer"
  target="_blank"
>
  <span>
    text
  </span>
</button>
`;

exports[`Input secondary 1`] = `
<button
  class="flex rounded-full transition-all font-semibold duration-200 ease-in shadow hover:shadow-md focus:outline-none focus:shadow-outline disabled:cursor-not-allowed disabled:opacity-75 px-4 py-2 text-sm border bg-white border-primary-500 text-primary-500 hover:bg-base-200 active:bg-base-300"
  rel="noreferrer"
  target="_blank"
>
  <span>
    text
  </span>
</button>
`;

exports[`Input secondary disabled 1`] = `
<button
  class="flex rounded-full transition-all font-semibold duration-200 ease-in shadow hover:shadow-md focus:outline-none focus:shadow-outline disabled:cursor-not-allowed disabled:opacity-75 px-4 py-2 text-sm border bg-white border-primary-500 text-primary-500 hover:bg-base-200 active:bg-base-300"
  disabled=""
  rel="noreferrer"
  target="_blank"
>
  <span>
    text
  </span>
</button>
`;
